source/expectations/data_values/ut_compound_data_value.tpb

```diff
@@ -229,63 +251,80 @@ create or replace type body ut_compound_data_value as
     l_column_filter := ut_compound_data_helper.get_columns_filter(a_exclude_xpath, a_include_xpath);
       
     /**
-    * Due to incompatibility issues in XML between 11 and 12.2 and 12.1 versions we will prepopulate pk_hash upfront to
-    * avoid optimizer incorrectly rewrite and causing NULL error or ORA-600
-    **/        
+    * Due to incompatibility issues in XML between 11 and 12.2 and 12.1 versions we will prepopulate item_hash upfront to
+    * avoid optimizer incorrectly rewrite and causing NULL error or ORA-600.
+    * We will also update item_data by include exclude values so we dont have to apply transformation over and over later.
+    **/     
     execute immediate 'merge into ' || l_ut_owner || '.ut_compound_data_tmp tgt
                        using (
                               select '||l_ut_owner ||'.ut_compound_data_helper.get_hash(ucd.item_data.getclobval()) item_hash, 
-                                      pk_hash, ucd.item_no, ucd.data_id
+                                      ucd.item_no, ucd.data_id, ucd.item_data
                               from
                               (
-                              select '||l_column_filter||','||get_column_pk_hash(a_join_by_xpath)||', item_no, data_id
+                              select '||l_column_filter||', item_no, data_id
                               from  ' || l_ut_owner || q'[.ut_compound_data_tmp ucd
                               where data_id = :self_guid or data_id = :other_guid
                               ) ucd
                        ) src
                        on (tgt.item_no = src.item_no and tgt.data_id = src.data_id)
                        when matched then update
-                       set tgt.item_hash = src.item_hash,
-                           tgt.pk_hash = src.pk_hash ]'
-                       using a_exclude_xpath, a_include_xpath,a_join_by_xpath,self.data_id, l_other.data_id;
+                       set tgt.item_hash = src.item_hash]'
+                       using a_exclude_xpath, a_include_xpath,
+                             self.data_id, l_other.data_id;
     
     /* Peform minus on two sets two get diffrences that will be used later on to print results */
-    execute immediate 'insert into ' || l_ut_owner || '.ut_compound_data_diff_tmp ( diff_id,item_hash,pk_hash,duplicate_no)
-                       with source_data as
-                       ( select t.data_id,t.item_hash,row_number() over (partition by t.pk_hash,t.item_hash,t.data_id order by 1,2) duplicate_no,
-                           pk_hash
+    execute immediate 'insert into ' || l_ut_owner || '.ut_compound_data_diff_tmp ( diff_id,item_hash,duplicate_no,item_data)
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Add a WHEN OTHERS exception handler to ensure that all exceptions are trapped. ([plsql:ExecuteImmediateTrapExceptionsCheck](https://sonarcloud.io/organizations/utplsql/rules#rule_key=plsql%3AExecuteImmediateTrapExceptionsCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=utPLSQL&pullRequest=760&issues=AWY_N4wenzfF1D4JsH2K&open=AWY_N4wenzfF1D4JsH2K)

[ISSUE_KEY]: AWY_N4wenzfF1D4JsH2K